libraries/ESP_NOW/src/ESP32_NOW_Serial.cpp

```diff
@@ -70,8 +70,25 @@ bool ESP_NOW_Serial_Class::begin(unsigned long baud) {
     //xSemaphoreTake(tx_sem, 0);
     xSemaphoreGive(tx_sem);
   }
-  setRxBufferSize(1024);  //default if not preset
-  setTxBufferSize(1024);  //default if not preset
+
+  size_t buf_size = 0;
+  if (ESP_NOW.getVersion() == 2) {
+    // ESP-NOW v2.0 has a larger maximum data length, so we need to increase the buffer sizes
+    // to hold around 3-4 packets
+    buf_size = setRxBufferSize(4096);
+    buf_size &= setTxBufferSize(4096);
```

Comment:
Maybe we should keep the 1024 buffer size but print a warning that the buffers might be too small.

Reply:
It's OK to have a bigger buffer. Let's not bother users with unnecessary warnings